packages/pages/src/upgrade/pagesUpdater.ts

```diff
@@ -43,38 +62,156 @@ const processDirectoryRecursively = (currentPath: string) => {
     const fileExtension = path.extname(filePath);
 
     if (isDirectory && !isGitDirectory && !isNodeModulesDirectory) {
-      processDirectoryRecursively(filePath);
+      processDirectoryRecursively(filePath, operation);
     } else if (
       [".js", ".ts", ".tsx"].includes(fileExtension) &&
       !isNodeModulesDirectory
     ) {
-      replaceImports(filePath);
+      try {
+        operation(filePath);
+      } catch (e) {
+        console.error(`Error processing ${filePath}: `, (e as Error).message);
+      }
     }
   });
+}
+
+/**
+ * Update sitesComponents to the latest version
+ * @param targetDirectory
+ */
+export const updateSitesComponents = async (targetDirectory: string) => {
+  await updatePackageDependency(
+    targetDirectory,
+    "@yext/sites-components",
+    null
+  );
+};
+
+/**
+ * Update pagesComponents to the latest version
```

Comment:
Give this a better description with example (since this will combine the two methods now given my other comment).